python/dolfinx/fem/assemble.py

```diff
@@ -297,10 +298,10 @@ def _assemble_matrix_csr(
 
 
 def apply_lifting(
-    b: np.ndarray,
+    b: npt.NDArray,
```

Comment:
Same comment as above for `b` and `x0`.

Reply:
Should I also revert `set_bc` in the same file?

Reply:
Let's try to minimize changes in this PR. Let's only revert those stated, and keep others (even if inconsistent, and open another PR to change them later on).